source/extensions/filters/udp/udp_proxy/router/router.h

```diff
@@ -0,0 +1,46 @@
+#pragma once
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "envoy/common/pure.h"
+#include "envoy/network/address.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace UdpFilters {
+namespace UdpProxy {
+namespace Router {
+
+/**
+ * The router which holds all clusters and determines the cluster which UDP data should be routed
+ * to.
+ */
+class Router {
+public:
+  virtual ~Router() = default;
+
+  /**
+   * Based on the address of the incoming UDP data, determine the target route for the data.
+   * @return the cluster name or empty string if there is not matching route for the data.
+   */
+  virtual const std::string
+  route(Network::Address::InstanceConstSharedPtr destination_address,
+        Network::Address::InstanceConstSharedPtr source_address) const PURE;
```

Comment:
nit: I would pass by const ref to avoid shared_ptr inc/dec

Reply:
Ok, I will make them `const`.
By the way, `InstanceConstSharedPtr` is a `shared_ptr<const Instance>`. I think it is not necessary to change to a const ref since the `const Instance` cannot be changed.